lightning/restore/restore.go

```diff
@@ -357,43 +358,37 @@ func (rc *RestoreController) listenCheckpointUpdates() {
 
 		lock.Unlock()
 
-		// gofail: var FailIfImportedChunk int
-		// if merger, ok := scp.merger.(*ChunkCheckpointMerger); ok && merger.Checksum.SumKVS() >= uint64(FailIfImportedChunk) {
-		// 	rc.checkpointsWg.Done()
-		// 	rc.checkpointsWg.Wait()
-		// 	panic("forcing failure due to FailIfImportedChunk")
-		// }
-		// goto RETURN1
-
-		// gofail: RETURN1:
-
-		// gofail: var FailIfStatusBecomes int
-		// if merger, ok := scp.merger.(*StatusCheckpointMerger); ok && merger.EngineID >= 0 && int(merger.Status) == FailIfStatusBecomes {
-		// 	rc.checkpointsWg.Done()
-		// 	rc.checkpointsWg.Wait()
-		// 	panic("forcing failure due to FailIfStatusBecomes")
-		// }
-		// goto RETURN2
-
-		// gofail: RETURN2:
-
-		// gofail: var FailIfIndexEngineImported int
-		// if merger, ok := scp.merger.(*StatusCheckpointMerger); ok && merger.EngineID == wholeTableEngineID && merger.Status == CheckpointStatusIndexImported && FailIfIndexEngineImported > 0 {
-		// 	rc.checkpointsWg.Done()
-		// 	rc.checkpointsWg.Wait()
-		// 	panic("forcing failure due to FailIfIndexEngineImported")
-		// }
-		// goto RETURN3
+		failpoint.Inject("FailIfImportedChunk", func(val failpoint.Value) {
+			if merger, ok := scp.merger.(*ChunkCheckpointMerger); ok && merger.Checksum.SumKVS() >= uint64(val.(int)) {
```

Comment:
🤔.

I wonder if there's plan to support

```go
failpoint.Inject("FailIfImportedChunk", func(val uint64) {
```

to avoid such awkward cast.

Reply:
It's hard to implement it. 😞